signal-hook-tokio/src/lib.rs

```diff
@@ -4,16 +4,57 @@
 
 //! A crate for integrating signal handling with the Tokio runtime.
 //!
-//! There are different sub modules for supporting different Tokio
-//! versions. The support for a version must be activated by a
-//! feature flag:
+//! This provides the [`Signals`][Signals] struct which acts as a
+//! [`Stream`][`futures::stream::Stream`] of signals.
 //!
-//! * `support-v0_1` for sub module [`v0_1`]
-//! * `support-v0_3` for sub module [`v0_3`]
+//! # Example
 //!
-//! See the specific sub modules for usage examples.
+//! ```rust
+//! use std::io::Error;
+//!
+//! use signal_hook::consts::signal::*;
+//! use signal_hook_tokio::Signals;
+//!
+//! use futures::stream::StreamExt;
+//!
+//! async fn handle_signals(signals: Signals) {
+//!     let mut signals = signals.fuse();
+//!     while let Some(signal) = signals.next().await {
+//!         match signal {
+//!             SIGHUP => {
+//!                 // Reload configuration
+//!                 // Reopen the log file
+//!             }
+//!             SIGTERM | SIGINT | SIGQUIT => {
+//!                 // Shutdown the system;
+//!             },
+//!             _ => unreachable!(),
+//!         }
+//!     }
+//! }
+//!
+//! #[tokio::main]
+//! async fn main() -> Result<(), Error> {
+//!     let signals = Signals::new(&[
+//!         SIGHUP,
+//!         SIGTERM,
+//!         SIGINT,
+//!         SIGQUIT,
+//!     ])?;
+//!     let handle = signals.handle();
+//!
+//!     let signals_task = tokio::spawn(handle_signals(signals));
+//!
+//!     // Execute your main program logic
+//!
+//!     // Terminate the signal stream.
+//!     handle.close();
+//!     signals_task.await?;
+//!
+//!     Ok(())
+//! }
+//! ```
 
-#[cfg(any(feature = "support-v0_1", feature = "support-v0_3"))]
 macro_rules! implement_signals_with_pipe {
```

Comment:
Is there any value in keeping this macro at this point, or should I go ahead and unwind it?

Reply:
I'll look into it in a follow-up PR.